source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -86,7 +86,7 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
     return Http::FilterHeadersStatus::Continue;
   }
 
-  if (!config_->filterEnabled(decoder_callbacks_->streamInfo().dynamicMetadata())) {
+  if (!config_->filterEnabled(route->routeEntry())) {
```

Comment:
no, please do not do this, it will break Istio and possibility other usages.

the documentation and the API is not wrong, it is supposed to be the dynamic metadata. If there is any mismatch, the doc should be updated instead of the implementation.

Reply:
I agree with @yangminzhu here.

Reply:
Hi @yangminzhu , thanks for checking this out, I should have created an issue to discuss this first... I definitely do not want to break anything. 

My understanding of dynamic metadata, is that it's used to share information between filters. Therefore, to have the dynamic metadata set for the `ext_authz` filter, it must be set by another filter? How does Istio work in this case?

Reply:
Agreed that the existing behavior is canon and we should update the docs, if anything. To @yangminzhu 's point, you can use per route config to disable ext_authz on a route instead of metadata.

re: Istio, my guess is that there are exist filters that set dynamic metadata to disable ext_authz based on custom logic.

Reply:
yes, In Istio we use a separate RBAC filter to generate the dynamic metadata, that allows us to implement the per-request enable/disable in a much more flexible way by reusing the existing RBAC API. You can find more details in this [design](https://docs.google.com/document/d/1V4mCQCw7mlGp0zSQQXYoBdbKMDnkPOjeyUb85U07iSI/edit#bookmark=kix.jdq8u0an2r6s).

Reply:
Hi @esmet, yes, I have seen the per route config, however, we are trying do something slightly different. 

@yangminzhu thanks, for sharing the document. This also works for what we're trying to do. 

Just to give you an example of how I understood the documentation : 

`
routes:
  - match:
      path: "/admin"
    metadata:
      filter_metadata:
        envoy.filters.http.ext_authz:
          some-key: "istio-ext-auth[0]"
  - match:
      path: "/token"
    metadata:
      filter_metadata:
        envoy.filters.http.ext_authz:
          some-key: "istio-ext-auth[1]"

#ext_auth filter 0
http_service:
    <omitted>
filter_enabled_metadata:
    filter: "envoy.filters.http.ext_authz"
    path:
      - key: "some-key"
    value:
      string_match:
        prefix: "istio-ext-auth[0]

#ext_auth filter 1
http_service:
    <omitted>
filter_enabled_metadata:
    filter: "envoy.filters.http.ext_authz"
    path:
      - key: "some-key"
    value:
      string_match:
        prefix: "istio-ext-auth[1]
`